### PR TITLE
Update NPM Dependencies and Refactor Code

### PR DESCRIPTION
**Pull request by jallentxbiomed**
- Added @labkey/build and upgraded @labkey/components
- Removed webpack files and migrated build scripts to newer labkey standards
- Moved entryPoints
- Removed unnecessary Client code for SndEventViewer
- Updated jest setup config file
- Updated react tests and snapshots to run with new dependency stack

### CODE REVIEW

File: snprc_ehr/package.json [hunk 1]
```diff
@@ -3,122 +3,58 @@
...
-    "setup": "npm install",
-    "build": "npm run clean && better-npm-run build:prod",
-    "start": "better-npm-run build:watch",
-    "build-prod": "npm run build",
-    "clean": "better-npm-run clean",
+    "setup": "npm ci --legacy-peer-deps",
+    "build": "npm run build-dev",
+    "build-dev": "npm run clean && cross-env NODE_ENV=development webpack --config node_modules/@labkey/build/webpack/dev.config.js --color",
+    "build-prod": "npm run clean && cross-env NODE_ENV=production webpack --config node_modules/@labkey/build/webpack/prod.config.js --color --progress --profile",
+    "clean": "rimraf resources/web/snprc_ehr/gen && rimraf resources/views/gen && rimraf resources/web/gen",
```
Comment (spamhurts) [inline snprc_ehr/package.json]: It doesn't look like we are generating code into "resources/web/snprc_ehr/gen" any longer. Can you verify?

File: snprc_ehr/package.json [hunk 2]
```diff
@@ -3,122 +3,58 @@
...
-    "react-router": "^3.2.6",
-    "react-select": "^3.1.0"
+    "@labkey/components": "2.242.9"
   },
   "devDependencies": {
-    "@babel/core": "^7.11.5",
-    "@babel/plugin-proposal-class-properties": "^7.10.4",
-    "@babel/plugin-proposal-object-rest-spread": "^7.11.0",
-    "@babel/plugin-transform-runtime": "^7.11.5",
+    "@babel/plugin-transform-runtime": "7.11.5",
```
Comment (spamhurts) [inline snprc_ehr/package.json]: Why are we removing the "^" from the package version? We probably want it to specify we don't want to accept major updates and that minor/patches are okay.

File: snprc_ehr/src/client/SsrsReporting/SsrsReporting.tsx [hunk 3]
```diff
@@ -6,7 +6,7 @@ import "./styles/ssrsReporting.scss";
 import { Button } from "react-bootstrap";
 import ReportSelectionPanel from "./components/ReportSelectionPanel";
 import InfoPanel from "../Shared/components/InfoPanel";
-import { OptionsType } from "react-select";
+import { OptionsType } from 'react-select';
```
Comment (spamhurts) [inline snprc_ehr/src/client/SsrsReporting/SsrsReporting.tsx]: necessary change?

Comment (jallentxbiomed) [reply snprc_ehr/src/client/SsrsReporting/SsrsReporting.tsx]: Nope, I ended up changing the version number of @labkey/components back and forth a few times and had to change this OptionsType in the process so I must have manually added this line back in.

File: snprc_ehr/package.json (hunk 2 above)
Comment (jallentxbiomed) [reply snprc_ehr/package.json]: Not quite sure, I'll add these back if you think we need them

File: snprc_ehr/package.json (hunk 1 above)
Comment (jallentxbiomed) [reply snprc_ehr/package.json]: Not on my end either, I'll remove that part of the clean script

File: snprc_ehr/package.json (hunk 2 above)
Comment (spamhurts) [reply snprc_ehr/package.json]: Thanks - I think we want them

**Review by spamhurts — APPROVED** [review]
Looks good

**Pull request merged**